nostrpostrlib/src/main/java/nostr/postr/Client.kt

```diff
@@ -24,23 +24,24 @@ object Client: RelayPool.Listener {
     internal var relays = Constants.defaultRelays
 
     fun connect(
+        subscriptionId: String = "main-channel",
         filters: MutableList<JsonFilter> = mutableListOf(JsonFilter()),
         relays: Array<Relay> = Constants.defaultRelays
     ) {
         this.filters = filters
         this.relays = relays
         RelayPool.register(this)
-        RelayPool.connect()
+        RelayPool.connect(subscriptionId)
     }
 
     fun disconnect() {
         RelayPool.unregister(this)
         RelayPool.disconnect()
     }
 
-    fun addFilter(filter: JsonFilter) {
+    fun addFilter(requestId: String = "main", filter: JsonFilter) {
```

Comment:
I think, addFilter was a bad idea.

On the Client level, useful functions needing changes around `subscriptionId` would be:

```
/**
 * Connect to the provided relays. Repeated calls gracefully connect to newly added relays and disconnect from removed relays.
 **/
fun connect(relays: Array<Relay> = Constants.defaultRelays)

/**
 * Disconnect from all relays
 **/
fun disconnect()

/**
 * Update subscription with the given subscriptionId. Automatically closes the subscription as soon as EOSE is received or 60s after request.
 **/
fun request(filters: List<JsonFilter>, subscriptionId: String = UUID.randomUUID().toString().substring(0..10))

/**
 * Update subscription with the given subscriptionId and watch until close() is called.
 **/
fun requestAndWatch(filters: List<JsonFilter>, subscriptionId: String)

fun close(subscriptionId: String)

override fun onEvent(event: Event, subscriptionId: String, relay: Relay) 

override fun onNewEvent(event: Event, subscriptionId: String)

...
```